pyproject.toml

```
@@ -25,67 +26,67 @@ packages = [
 ]
 
 [tool.poetry.dependencies]
-python = "^3.7"
+python = ">=3.7,<3.11"
 click = ">=7.1<=9.0"
-click-default-group = "*"
-colour = "*"
+click-default-group = "^1.2.2"
+colour = "^0.1.5"
 numpy = "^1.9"
-Pillow = "*"
-scipy = "*"
-tqdm = "*"
-pydub = "*"
-pygments = "*"
+Pillow = "^8.4.0"
+scipy = "^1.7.3"
+tqdm = "^4.62.3"
+pydub = "^0.25.1"
 rich = ">=6.0"
 pycairo = "^1.19"
 manimpango = "^0.3.0"
 networkx = "^2.5"
 decorator = "^5.0.7"
-setuptools = "*"
-importlib-metadata = {version = "*", python = "<3.8"}
-grpcio =  { version = "1.33.*", optional = true }
-grpcio-tools = { version = "1.33.*", optional = true }
-watchdog = "*"
+setuptools = "^60.1.0"
+importlib-metadata = { version = "^4.10.0", python = "<3.8" }
+grpcio =  { version = "^1.43.0", optional = true }
+grpcio-tools = { version = "^1.43.0", optional = true }
+watchdog = "^2.1.6"
 jupyterlab = { version = "^3.0", optional = true }
 moderngl = "^5.6.3"
 moderngl-window = "^2.3.0"
 mapbox-earcut = "^0.12.10"
 cloup = "^0.7.0"
-requests = "*"
+requests = "^2.26.0"
 dearpygui = { version = "^0.8", optional = true }
 screeninfo = "^0.6.7"
 skia-pathops = "^0.7.0"
 isosurfaces = "0.1.0"
 srt = "^3.5.0"
+Pygments = "^2.10.0"
 
 [tool.poetry.extras]
 webgl_renderer = ["grpcio","grpcio-tools"]
 jupyterlab = ["jupyterlab"]
 gui = ["dearpygui"]
 
 [tool.poetry.dev-dependencies]
-pytest-cov = "*"
+pytest-cov = "^3.0.0"
 pytest = "^6.0"
-pylint = "*"
+pylint = "^2.12.2"
 Sphinx = "^4"
-sphinx-copybutton = "*"
-sphinxext-opengraph = "*"
-furo = "*"
-recommonmark = "*"
+sphinx-copybutton = "^0.4.0"
+sphinxext-opengraph = "^0.5.1"
+furo = "^2021.11.23"
+recommonmark = "^0.7.1"
 matplotlib = "^3.3.2"
 pre-commit = "^2.11.1"
 gitpython = "^3"
 pygithub = "^1"
 flake8 = "^3.9.0"
 isort = "^5.8.0"
 pytest-xdist = "^2.2"
-psutil = ">=3.0"
 flake8-builtins = "^1.5.3"
 flake8-bugbear = "^21.4.3"
 flake8-docstrings = "^1.6.0"
 flake8-rst-docstrings = "^0.2.3"
 flake8-pytest-style = "^1.5.0"
 flake8-simplify = "^0.14.1"
 flake8-comprehensions = "^3.7.0"
+psutil-wheels = "^5.8.1"
```

Poetry doesn't support yanked releases, https://github.com/python-poetry/poetry/issues/2453.
```suggestion
psutil-wheels = "5.8.0"
```